app/src/main/java/com/waseefakhtar/doseapp/MainActivity.kt

```diff
@@ -7,21 +7,26 @@ import androidx.activity.compose.setContent
 import com.waseefakhtar.doseapp.analytics.AnalyticsEvents
 import com.waseefakhtar.doseapp.analytics.AnalyticsHelper
 import dagger.hilt.android.AndroidEntryPoint
+import javax.inject.Inject
 
 @AndroidEntryPoint
 class MainActivity : ComponentActivity() {
+
+    @Inject
+    lateinit var analyticsHelper: AnalyticsHelper
```

Comment:
Maybe could be important inject AnalyticsHelper using Hilt DSL for Compose screens/components instead inject directly on activity.

I think this because on your approuch you will need pass the analytics value as a cascate to anothers components and screens and one of the DI ideia is: Not depend from another components to receive the value, you could recover the value from anywhere, injecting the value.

My suggestion could be strange here because the AnalyticsHelper wasn't used on ViewModels, and may the best approach is migrate AnalyticsHelper to ViewModels instead. 

What do you think about it?

Reply:
I considered this approach as well, and that's why I initially discussed it with @waseefakhtar here: https://github.com/waseefakhtar/dose-android/issues/74#issuecomment-1807153025.

Given the project's current structure, this was my initial approach to refactoring. I remain open to suggestions, and if this approach proves unsuitable, I can always revert to the second option mentioned in the comment link.

Reply:
Nice! I didn't know about the approach dicussed, thanks for showing to me @ankur141295!

Reply:
I agree with the approach of migrating AnalyticsHelper to ViewModels instead! That'd make the code a ton simpler and easy to test. 

@ankur141295 my bad regarding the discussion. I might've missed your point there.

Reply:
@waseefakhtar No worries, I'll make the code changes again and inject AnalyticsHelper into ViewModel instead.